### PR TITLE
Add Windows .bat launcher and setup

### PR DESCRIPTION
**Pull request by sirloinofbeef**
.bat scripts for Windows users to: 
1) Install requirements.txt and present options to configure scheduled tasks (Windows equivalent of cron job) and Windows shortcuts
2) Check for Python and launch plex_trakt_sync

The purpose of these optional Windows scripts is to provide an application-like arrangement for less tech savvy Windows users seeking a more 'point and click' style interface.

### CODE REVIEW

File: main.bat [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+:: Check for Python Installation
+py --version 2>NUL
+if errorlevel 1 goto errorNoPython
+
+cls
+
+title Plex - Trakt Sync
+
+
+py main.py
```
Comment (glensc) [inline main.bat:19]:
```suggestion
py -m plex_trakt_sync
```

`main.py` is deprecated, should be using module mode:
- https://github.com/Taxel/PlexTraktSync#setup

you should also rename your script accordingly.

**Comment by glensc** [comment]
also fix PR title

File: main.bat (hunk 1 above)
Comment (glensc) [reply main.bat:19]: the wrapper must also accept extra arguments, for example `py -m plex_trakt_sync watch`

**Comment by glensc** [comment]
this script is in no way "simple", update PR body accordingly

**Comment by sirloinofbeef** [comment]
If people are happy with this I will code a 'setup' .bat for Windows as well which will install requirements.txt and present options to configure scheduled tasks etc.

File: setup.bat [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+IF /I "%CONFIGCHOICE%"=="1" GOTO :RESET
+IF /I "%CONFIGCHOICE%"=="3" GOTO :ADDSERVER
+GOTO :ENDIT
+
+:RESET
+schtasks /delete /tn "Plex Trakt Sync" /f >nul 2>&1
+del "%userprofile%\Start Menu\Programs\Startup\Plex Trakt Sync.lnk" /f >nul 2>&1
+del "%userprofile%\Desktop\Plex Trakt Sync.lnk" /f >nul 2>&1
+del ".env" /f >nul 2>&1
+del ".pytrakt.json" /f >nul 2>&1
```
Comment (glensc) [inline setup.bat:52]: what about trakt cache file?

File: setup.bat [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+IF /I "%STARTUP%" EQU "Y" echo.    * Automatically at system startup
+IF /I "%DESKTOPQ%" EQU "Y" echo.    * Manually using the shortcut on your desktop
+echo.    * Manually using the shortcut created at %cd%\Plex Trakt Sync
+echo.
+
+echo Press any key to continue with setup...
+echo.
+pause > nul
+echo Installing Python requirements...
+pip install -r data\requirements.txt
```
Comment (glensc) [inline setup.bat:93]: this project does not have such dir as "data"

**Comment by glensc** [comment]
please describe in the PR body what this script does, it no longer just launches things.

**Comment by twolaw** [comment]
@sirloinofbeef used a customised version of the script : https://github.com/sirloinofbeef/PlexTraktSync
The official version is now moving forward, so he kind of abandons his own fork.

The person at the origin of this copyright violation :
- https://github.com/Taxel/PlexTraktSync/pull/156#pullrequestreview-624173150

**Comment by sirloinofbeef** [comment]
> @sirloinofbeef used a customised version of the script : https://github.com/sirloinofbeef/PlexTraktSync
> The official version is now moving forward, so he kind of abandons his own fork, that's why he brings this `.bat` file here.
> 
> The person at the origin of this copyright violation :
> 
> * [Test #156 (review)](https://github.com/Taxel/PlexTraktSync/pull/156#pullrequestreview-624173150)

As stated previously I created the fork and added myself to the copyright because this project seemed abandoned. I was unaware at the time that work was continuing in a different branch. I have apologised for the mistake on my part and deprecated the fork.

File: setup.bat (hunk 2 above)
Comment (sirloinofbeef) [reply setup.bat:52]: Fixed. Thanks.

**Comment by glensc** [comment]
@sirloinofbeef what windows version does this requires? any other special requirements?

**Comment by glensc** [comment]
this trakt.ico, how was it produced, what is its license?

**Comment by sirloinofbeef** [comment]
> @sirloinofbeef what windows version does this requires? any other special requirements?

Should work from Windows 7 - Windows 11

**Comment by sirloinofbeef** [comment]
> this trakt.ico, how was it produced, what is its license?

Taken from https://trakt.tv/branding as per their guidelines.
We can develop our own unique icon if preferred though.

**Comment by glensc** [comment]
include that version info to PR body too.

also, this should have a section in the readme regarding windows launcher and installer. it should also explain why should one even use these wrappers, as python is cross-platform just run `python -m plex_trakt_sync`...


also, do note I can't support any of this .bat file magic if people come complaining...

**Comment by glensc** [comment]
are you sure `py` is present in earlier than windows 10?

from this issue to me it seemed it's win10 app store feature:
- https://github.com/Taxel/PlexTraktSync/issues/427

**Comment by sirloinofbeef** [comment]
> @sirloinofbeef what windows version does this requires? any other special requirements?

I have l

> are you sure `py` is present in earlier than windows 10?
> 
> from this issue to me it seemed it's win10 app store feature:
> 
> * [Permission denied error when using  ./plex_trakt_sync.sh watch for windows Fixed by using py instead of python3 #427](https://github.com/Taxel/PlexTraktSync/issues/427)

I have updated the code to use `python` for wider support, and clarified support to be from Win7sp1 - 11.

**Comment by sirloinofbeef** [comment]
> include that version info to PR body too.
> 
> also, this should have a section in the readme regarding windows launcher and installer. it should also explain why should one even use these wrappers, as python is cross-platform just run `python -m plex_trakt_sync`...
> 
> also, do note I can't support any of this .bat file magic if people come complaining...

Essentially this is about providing a more 'application' like experience for less tech savvy users who just want to point and click.

**Comment by glensc** [comment]
The correct capitalization is macOS not MacOS:
- https://en.wikipedia.org/wiki/macOS

**Comment by glensc** [comment]
also, WSL behaves like Linux:
- https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux

**Comment by sirloinofbeef** [comment]
> also, WSL behaves like Linux:
> 
> * https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux

Theoretically yes, you could run it under WSL. It would be a bit overkill though as it runs in native Windows fine.

**Comment by glensc** [comment]
point being that the section that you titled linux/macos can be used on windows too, so it's not accurate to exclude windows there. perhaps keep it "universal" 
 or something (or just keep it "installation") and have only windows installer specific section?

**Comment by glensc** [comment]
Removal of "(cross platform)" is also in order. but since that changing things to bulleted list is very large change, could you perhaps to that change in a separate merge request instead? just omit the new installer section details.

**Comment by sirloinofbeef** [comment]
> Removal of "(cross platform)" is also in order. but since that changing things to bulleted list is very large change, could you perhaps to that change in a separate merge request instead? just omit the new installer section details.

Agree that makes sense.

**Comment by glensc** [comment]
@sirloinofbeef please review your readme changes. as the diff view shows added for "Sync settings" means it's added twice:
- https://github.com/Taxel/PlexTraktSync/pull/486/files

**Pull request merged**